### PR TITLE
Update release checklist with bumping release version

### PR DESCRIPTION
**Pull request by hickeyma**
**What this PR does / why we need it**:
Add to the release checklist about updating the release version in the master branch when creating a release.  This is so that the correct release version is shown when build from master.

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: docs/release_checklist.md [hunk 1]
```diff
@@ -147,6 +147,24 @@ git add .
...
+
+# create new branch off master
+git checkout master
+git checkout -b bump-version-<release_version>
+
+# cherry pick the commit using id from first command
+git cherry-pick -x <commit-id>
+
+# commit the change
+git push upstream bump-version-<release-version>
```
Comment (bacongobbler) [inline docs/release_checklist.md]: In this particular case, we probably want to push it to your origin remote instead. The only time we want to push to upstream is for the release branches. :)

**Review by bacongobbler — COMMENTED** [review]
One minor thing but LGTM otherwise! Thanks for picking this up

**Review by bacongobbler — APPROVED** [review]

File: docs/release_checklist.md (hunk 1 above)
Comment (hickeyma) [reply docs/release_checklist.md]: Nice catch, thanks. *shaking fist* at copy/paste! :)

**Comment by hickeyma** [comment]
Thanks for review @bacongobbler

**Pull request merged**